### PR TITLE
evolution land

### PR DESCRIPTION
**Pull request by waynebruce0x**

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
 
    
        Unknown chain(s): crab
        Note: if you think that the chain is correct but missing from our list, please add it to 'projects/helper/chains.json' file

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
 
    
        Please move the following keys into the chain: tvl
    
        We have a new adapter export specification now where tvl and other chain specific information are moved inside chain export.
        For example if your protocol is on ethereum and has tvl and pool2, the export file would look like:
        
            module.exports = {
              methodlogy: '...',
              ethereum: {
                tvl: 
                pool2:
              }
            }

File: projects/evolutionland/index.js [hunk 1]
```diff
@@ -208,4 +261,8 @@ module.exports = {
   polygon: {
     tvl: polygonTvl
   },
+  crab: {
+    tvl: crabTvl
+  },
+  tvl: sdk.util.sumChainTvls([ethTvl, hecoTvl, polygonTvl, crabTvl])
```
Comment (g1nt0ki) [inline projects/evolutionland/index.js:267]: this is no longer needed right?

Comment (waynebruce0x) [reply projects/evolutionland/index.js:267]: yeah true

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
  
    
    
    Error: Couldn't find block height for chain crab, RPC node rugged
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:171:27)
        at step (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:44:23)
        at Object.throw (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:25:53)
        at rejected (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:17:65)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
  
    
    
    Error: Couldn't find block height for chain crab, RPC node rugged
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:171:27)
        at step (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:44:23)
        at Object.throw (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:25:53)
        at rejected (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:17:65)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
  
    
    
    Error: Couldn't find block height for chain crab, RPC node rugged
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:171:27)
        at step (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:44:23)
        at Object.throw (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:25:53)
        at rejected (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:17:65)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
  
    
    
    Error: Couldn't find block height for chain crab, RPC node rugged
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:171:27)
        at step (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:44:23)
        at Object.throw (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:25:53)
        at rejected (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:17:65)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

File: projects/milko-farm/index.js [hunk 2]
```diff
@@ -15,4 +15,4 @@ module.exports = {
     }
 }
 
-
+// node test.js projects/milko-farm/index.js
```
Comment (g1nt0ki) [inline projects/milko-farm/index.js:18]: I forgot to delete this file, can you delete it?

File: projects/milko-farm.js [hunk 3]
```diff
@@ -5,7 +5,12 @@ const masterchef = "0x5d0C5db1D750721Ed3b13a8436c17e035B44c3D0";
 const wada = "0xAE83571000aF4499798d1e3b0fA0070EB3A3E3F9";
 const whitelist = [
   "0x3c786134228b363fb2984619D7560AB56363B2bD",
-  "0xB44a9B6905aF7c801311e8F4E76932ee959c663C"
+  "0xB44a9B6905aF7c801311e8F4E76932ee959c663C", //
+  "0x6aB6d61428fde76768D7b45D8BFeec19c6eF91A8",
+  "0xAE83571000aF4499798d1e3b0fA0070EB3A3E3F9", //
+  "0xE3F5a90F9cb311505cd691a46596599aA1A0AD7D", //
+  "0x639A647fbe20b6c8ac19E48E2de44ea792c62c5C",
+  "0x80A16016cC4A2E6a2CACA8a4a498b1699fF0f844"
```
Comment (g1nt0ki) [inline projects/milko-farm.js:13]: I guess you will revert changes to this file

**Comment by llamatester** [comment]
Error while running adapter at projects/evolutionland/index.js: 
        
 
  
    
    
    Error: Couldn't find block height for chain crab, RPC node rugged
        at Object.<anonymous> (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:171:27)
        at step (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:44:23)
        at Object.throw (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:25:53)
        at rejected (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/node_modules/@defillama/sdk/build/util/index.js:17:65)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Pull request merged**